flask_appbuilder/tests/test_api.py

```diff
@@ -2788,3 +2837,137 @@ class Model1PermConverge(ModelRestApi):
         self.assertEqual(state_transitions, target_state_transitions)
         role = self.appbuilder.sm.find_role("Test")
         self.assertEqual(len(role.permissions), 5)
+
+    def test_before_request(self):
+        """
+            REST Api: Test simple before_request filter
+        """
+        client = self.app.test_client()
+        token = self.login(client, USERNAME_ADMIN, PASSWORD_ADMIN)
+
+        # Everything is disabled
+        self._before_request_condition_value = False
+        self._before_request_can_read = False
+        self._before_request_can_write = False
+
+        uri = "api/v1/model1beforerequest/basic"
```

Comment:
optional: we could do a for loop here to DRY these tests a bit

Reply:
Personally, I like keeping test code "damp" rather than DRY because I think that writing out each case explicitly is easier to read, serving as documentation. Also, less likely to introduce a bug into tests if you avoid "programming" in tests.

That being said, I am happy to change it if you feel strongly.